### ISSUE TITLE
How is the test data for TyDiQA generated?

### ISSUE THREAD

**Issue by namiyousef** [google-research/xtreme#95]
Hi,

As of 05.09.22, the GCS bucket for TyDiQA contains test sets that were translated from the target language to English. Naturally there is no en-en translation. 

However, it is unclear to me where these translations came from... do they come from the development set? The xtreme paper mentions 323-2719 examples for the test set, but the development set for tydiqa is 634 (as per xtreme). TyDiQA itself does not have a test set (see [repo](https://github.com/google-research-datasets/tydiqa)).

Also on huggingface, both tydiqa (see dataset [card](https://huggingface.co/datasets/tydiqa)) and xtreme-tydiqa (see dataset [card](https://huggingface.co/datasets/xtreme)) have 440 examples.

Could you please clarify this discrepancy?

**Comment by sebastianruder** [comment]
Hi,
Thanks a lot for the question. The GoldPassage task for TyDiQA consists of training and dev sets for each language. As there are no explicit test sets, we use the dev sets as test sets for XTREME (the listed numbers of 323-2719 examples pertain to these test sets). To align with the evaluation setting for the other QA tasks, we use only the English TyDiQA training data for training and the English TyDiQA dev data (consisting of 634 examples) for development. The training data of the other languages is thus not used in the standard setup. 

The translations in the [translate-test](https://pantheon.corp.google.com/storage/browser/xtreme_translations/TyDiQA-GoldP/translate-test?pageState=(%22StorageObjectListTable%22:(%22f%22:%22%255B%255D%22))&prefix=&forceOnObjectsSortingFiltering=false) folder contain the translations of the test sets (the original dev sets in the other languages) translated to English.

I hope things are clearer now. Let me know if you still have any questions.

**Comment by namiyousef** [comment]
Hi @sebastianruder,

Thank for you very much for the reply, it is clearer now.

I'm still unsure about some of the numbers presented in the paper. For one, I'm unsure where the 634 English examples for the development set come from. The development set for tydiqa downloaded alongside download_data.sh has 440 English instances only. Further, upon further examination of the dev set, I find that bengali and korean have less instances than the lower threshold reported in xtreme (323), with 276 and 113 instances respectively. The upper threshold (2719) also doesn't correspond to any languages in the dev set (the language with the highest number of instances is arabic with 921 instances).

With regards to translate-test, I don't need them for my current project but I was using them to check against the hugginface repos and the xtreme GCS data. I was surprised to find that their sizes don't match up with the tydiqa dev set, since as you say they should be translations of the dev set.

I'd greatly appreciate if you could clarify the above. Once that's done I'm happy to close this issue.

Also are you currently accepting any contributions to this repository?

Best,
Yousef

**Comment by sebastianruder** [comment]
Hi Yousef,

Thanks for looking at the data. It's been a while since I generated the data. I'm unfortunately not sure right now what is the source of the discrepancies. I'm a bit swamped right now but will take a closer look once I have some time. Sorry that I don't have a more satisfactory response at the moment.

Yes, we are accepting contributions.

Best,
Sebastian
